### PR TITLE
[v7r0]  removed unused/outdated stuff from Distribution module

### PR DESCRIPTION
**Pull request by fstagni**
BEGINRELEASENOTES

*Core
CHANGE: removed unused/outdated stuff from Distribution module

ENDRELEASENOTES

### CODE REVIEW

File: Core/Utilities/Distribution.py [hunk 1]
```diff
@@ -1,464 +1,86 @@
...
   if not verTup:
     return S_OK()
-  initFile = os.path.join( rootPath, "__init__.py" )
-  if not os.path.isfile( initFile ):
+  initFile = os.path.join(rootPath, "__init__.py")
+  if not os.path.isfile(initFile):
     return S_OK()
   try:
-    fd = open( initFile, "r" )
+    fd = open(initFile, "r")
```
Comment (andresailer) [inline Core/Utilities/Distribution.py]: `with open...`

File: Core/Utilities/Distribution.py [hunk 2]
```diff
@@ -1,464 +1,86 @@
...
+      if os.path.isdir(dirToTar):
+        tf.add(dirToTar, os.path.basename(os.path.abspath(dirToTar)), recursive=True)
   tf.close()
   md5FilePath = False
-  for suffix in ( ".tar.gz", ".gz" ):
-    sLen = len( suffix )
-    if tarballPath[ len( tarballPath ) - sLen: ] == suffix:
+  for suffix in (".tar.gz", ".gz"):
+    sLen = len(suffix)
+    if tarballPath[len(tarballPath) - sLen:] == suffix:
```
Comment (andresailer) [inline Core/Utilities/Distribution.py]: `if tarballPath.endswith(suffix):`

File: WorkloadManagementSystem/Agent/OptimizerModule.py [hunk 3]
```diff
@@ -44,258 +44,257 @@ def initialize( self, jobDB = False, logDB = False ):
       self.logDB = logDB
 
     trailing = "Agent"
-    optimizerName = self.am_getModuleParam( 'agentName' )
-    if optimizerName[ -len( trailing ):].find( trailing ) == 0:
-      optimizerName = optimizerName[ :-len( trailing ) ]
-    self.am_setModuleParam( 'optimizerName', optimizerName )
+    optimizerName = self.am_getModuleParam('agentName')
+    if optimizerName[-len(trailing):].find(trailing) == 0:
```
Comment (andresailer) [inline WorkloadManagementSystem/Agent/OptimizerModule.py]: Does this also just do `if optimizerName.endswith('Agent')`?

File: Core/Utilities/Distribution.py (hunk 1 above)
Comment (chaen) [reply Core/Utilities/Distribution.py]: Doesn't the distribution tools need to be python 2.6 compatible ?

Comment (fstagni) [reply Core/Utilities/Distribution.py]: No, the distribution tools should be OK with 2.7 only.

File: Core/Utilities/Distribution.py [hunk 4]
```diff
@@ -1,464 +1,85 @@
...
+    with open(initFile, "r") as fd:
+      fileData = fd.read()
   except Exception as e:
-    return S_ERROR( "Could not open %s: %s" % ( initFile, str( e ) ) )
-  versionStrings = ( "majorVersion", "minorVersion", "patchLevel", "preVersion" )
+    return S_ERROR("Could not open %s: %s" % (initFile, str(e)))
+  versionStrings = ("majorVersion", "minorVersion", "patchLevel", "preVersion")
   reList = []
-  for iP in range( len( versionStrings ) ):
+  for iP in enumerate(versionStrings):
```
Comment (andresailer) [inline Core/Utilities/Distribution.py]: enumerate gives you a iterator with pair of index and iterable entry.
So this has to be 
```
for iP, version in enumerate(versionStrings):
```

**Comment by fstagni** [comment]
Review OK?

File: Core/Utilities/Distribution.py [hunk 5]
```diff
@@ -1,464 +1,85 @@
...
-  fd = open( md5FilePath, "w" )
-  fd.write( md5str )
+    return S_ERROR("Could not generate md5 filename")
+  md5str = File.getMD5ForFiles([tarballPath])
+  fd = open(md5FilePath, "w")
+  fd.write(md5str)
   fd.close()
   return S_OK()
-
-#Start of release notes
```
Comment (atsareg) [inline Core/Utilities/Distribution.py]: Why do you delete the release notes related functions ?

Comment (fstagni) [reply Core/Utilities/Distribution.py]: Sorry, restored.

**Pull request merged**